### PR TITLE
Prevent loops in account creation

### PR DESCRIPTION
**Pull request by thecadams**

### CODE REVIEW

**Comment by thecadams** [comment]
Going to 🔨 🍷  and merge this, we can review later.

File: src/pages/Account/Profile.tsx [hunk 1]
```diff
@@ -44,8 +45,10 @@ const Profile: React.FunctionComponent<PropsType> = ({ location }) => {
   const onSubmit = async (data: ProfileFormData) => {
     if (!user) return;
     await updateUserProfile(user.uid, data);
-    const accountQuestionsUrl = `/account/questions?venueId=${venueId}&returnUrl=${window.location.pathname}${window.location.search}`;
-    const nextUrl = venueId ? accountQuestionsUrl : `/${DEFAULT_VENUE}`;
```
Comment (thecadams) [inline src/pages/Account/Profile.tsx:48]: These are the core of the change. `returnUrl` should never take the user back to part of the account creation flow; that's a recipe for loops.

And the `DEFAULT_VENUE` is going to confusingly send _someone_ to the wrong venue at some point, so it needs to go.

File: src/pages/Account/Profile.tsx [hunk 2]
```diff
@@ -44,8 +45,10 @@ const Profile: React.FunctionComponent<PropsType> = ({ location }) => {
...
+    const accountQuestionsUrl = `/account/questions?venueId=${venueId}${
+      returnUrl ? "&returnUrl=" + returnUrl : ""
+    }`;
```
Comment (thecadams) [inline src/pages/Account/Profile.tsx:50]: Bringing `returnUrl` along through the flow here.

File: src/pages/VenuePage/VenuePage.tsx [hunk 3]
```diff
@@ -295,7 +295,7 @@ const VenuePage = () => {
     return <LoadingPage />;
   }
 
-  if (profile && !(profile?.partyName && profile?.pictureUrl)) {
+  if (profile && !isCompleteProfile(profile)) {
```
Comment (thecadams) [inline src/pages/VenuePage/VenuePage.tsx:298]: `isCompleteProfile` is an attempt to make this code a little more self documenting.

File: src/hooks/useUser.ts [hunk 4]
```diff
@@ -1,7 +1,14 @@
...
+import { FirebaseReducer } from "react-redux-firebase";
+import { User } from "types/User";
 
-export const useUser = () => {
+type UseUserResult = {
+  user: FirebaseReducer.AuthState | undefined;
+  profile: FirebaseReducer.Profile<User> | undefined;
+};
+
+export const useUser = (): UseUserResult => {
```
Comment (thecadams) [inline src/hooks/useUser.ts:11]: `profile` was coming through this hook as `any` type. So this was overdue.

File: src/pages/Account/Profile.tsx (hunk 2 above)
Comment (0xdevalias) [reply src/pages/Account/Profile.tsx:50]: [Future] We can use https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams and similar to handle this in a better way than string concatenation.

File: src/pages/Account/Profile.tsx [hunk 5]
```diff
@@ -44,8 +45,10 @@ const Profile: React.FunctionComponent<PropsType> = ({ location }) => {
...
+    const nextUrl = venueId ? accountQuestionsUrl : returnUrl?.toString() ?? "";
     history.push(IS_BURN ? `/enter/step3` : nextUrl);
```
Comment (0xdevalias) [inline src/pages/Account/Profile.tsx:52]: [Future] We should be stripping out all of this `IS_BURN` logic from this flow as well.

File: src/pages/VenuePage/VenuePage.tsx (hunk 3 above)
Comment (0xdevalias) [reply src/pages/VenuePage/VenuePage.tsx:298]: Excellent! This is what I wanted in my retroactive review of https://github.com/sparkletown/sparkle/pull/932#pullrequestreview-547762211

**Review by 0xdevalias — COMMENTED** [review]
Retroactive review, but this looks solid!

**Pull request merged**